diff --git a/og_menu.js b/og_menu.js
index cd6e04e..ab1b309 100644
--- a/og_menu.js
+++ b/og_menu.js
@@ -7,17 +7,17 @@
 Drupal.ogMenu = Drupal.ogMenu || {};
 
 Drupal.behaviors.ogMenuGroupswitch = {
-    attach: function(context) {
-      // Initialize variables and form.
-      Drupal.ogMenu.originalParent = $('.menu-parent-select').val(); // Get original parent. We'll use this shortly.
-      Drupal.ogMenu.selected = []; // Create Variable to hold selected groups
-      Drupal.ogMenu.bindEvents(); // Bind events to group audience fields.
-      Drupal.ogMenu.setSelected(); // Get all currently selected.
-      Drupal.ogMenu.populateParentSelect(); // Populate
-
-      // Make sure the originalParent is set on page load.
-      $('.menu-parent-select').val(Drupal.ogMenu.originalParent);
-    }
+  attach: function(context) {
+    // Initialize variables and form.
+    Drupal.ogMenu.originalParent = $('.menu-parent-select').val(); // Get original parent. We'll use this shortly.
+    Drupal.ogMenu.selected = []; // Create Variable to hold selected groups
+    Drupal.ogMenu.bindEvents(); // Bind events to group audience fields.
+    Drupal.ogMenu.setSelected(); // Get all currently selected.
+    Drupal.ogMenu.populateParentSelect(); // Populate
+
+    // Make sure the originalParent is set on page load.
+    $('.menu-parent-select').val(Drupal.ogMenu.originalParent);
+  }
 };
 
 /**
@@ -27,7 +27,7 @@ Drupal.ogMenu.bindEvents = function() {
   var selector = '';
   $.each(Drupal.settings.ogMenu.group_audience_fields, function (index, value) {
     // Only bind events to visible fields.
-    if (value.visible === true) {
+    if (value.visibility === true) {
       selector = Drupal.ogMenu.buildSelector(index, value.normal, value.cardinality);
       Drupal.ogMenu.bindEvent(value.normal, selector, value.cardinality);
       if (Drupal.settings.ogMenu.administer_group === true) {
@@ -198,13 +198,13 @@ Drupal.ogMenu.populateParentSelect = function() {
   });
 
   var parentToSetActive = Drupal.ogMenu.selected[0];
-  activeIsSet = 0;
+  var activeIsSet = 0;
 
   // Add any og_menus to the menu-parent-select menu
   $.each(Drupal.settings.ogMenu.menus, function(menu_name, gid) {
     if ($.inArray(gid,Drupal.ogMenu.selected) >= 0)  {
       $.each(Drupal.settings.ogMenu.parent_options, function(key,val) {
-        parts = key.split(':');
+        var parts = key.split(':');
         if (parts[0] === menu_name) {
           if (gid == parentToSetActive && activeIsSet === 0) {
             // Add option to Select and set as selected.
diff --git a/og_menu.module b/og_menu.module
index 9b4f64d..26273db 100644
--- a/og_menu.module
+++ b/og_menu.module
@@ -364,20 +364,18 @@ function og_menu_form_node_form_alter(&$form, &$form_state) {
       foreach (og_get_group_audience_fields('node', $type) as $field_name => $label) {
         $field = field_info_field($field_name);
         $instance = field_info_instance('node', $field_name, $type);
-        // This is currently the only widget in OG.
-        if ($instance['widget']['type'] == 'og_complex') {
-          // Group the data we need in the client side js code.
-          // Notice: 'default' is a reserved word in js, so we use 'normal'
-          // instead.
-          $fields[$field_name]['normal'] = $instance['settings']['behaviors']['og_widget']['default']['widget_type'];
-          $fields[$field_name]['admin'] = $instance['settings']['behaviors']['og_widget']['admin']['widget_type'];
-          $fields[$field_name]['cardinality'] = $field['cardinality'];
-          // Field is visible by default.
-          $fields[$field_name]['visibility'] = TRUE;
-          // When using entity reference pre-populate, the field may be hidden.
-          if (isset($instance['settings']['behaviors']['prepopulate']['action']) && $instance['settings']['behaviors']['prepopulate']['action'] == 'hide' && $context['group_type'] == $field['settings']['target_type']) {
-            $fields[$field_name]['visibility'] = $context['gid'];
-          }
+
+        // Group the data we need in the client side js code.
+        // Notice: 'default' is a reserved word in js, so we use 'normal'
+        // instead.
+        $fields[$field_name]['normal'] = $instance['settings']['behaviors']['og_widget']['default']['widget_type'];
+        $fields[$field_name]['admin'] = $instance['settings']['behaviors']['og_widget']['admin']['widget_type'];
+        $fields[$field_name]['cardinality'] = $field['cardinality'];
+        // Field is visible by default.
+        $fields[$field_name]['visibility'] = TRUE;
+        // When using entity reference pre-populate, the field may be hidden.
+        if (isset($instance['settings']['behaviors']['prepopulate']['action']) && $instance['settings']['behaviors']['prepopulate']['action'] == 'hide' && $context['group_type'] == $field['settings']['target_type']) {
+          $fields[$field_name]['visibility'] = $context['gid'];
         }
       }
       $settings['group_audience_fields'] = $fields;
