diff --git a/og_menu.js b/og_menu.js
index d51e998..8bc3902 100644
--- a/og_menu.js
+++ b/og_menu.js
@@ -11,7 +11,7 @@
       var enabled = $('#edit-menu-enabled').is(':checked');
 
       var holder = document.createElement('select');
-      var selectors = 'select[name^="og_group_ref"], input[name^="og_group_ref"';
+      var selectors = 'select[name^="og_group_ref"], input[name^="og_group_ref"]';
 
       // Toggle menu alteration
       function toggle(values) {
diff --git a/og_menu.module b/og_menu.module
index 2abc833..4664ad2 100644
--- a/og_menu.module
+++ b/og_menu.module
@@ -952,8 +952,8 @@ function _og_menu_autocomplete($string) {
 function og_menu_get_node_groups($group_type, $object) {
   $groups = og_get_entity_groups($group_type, $object);
 
-  if (!empty($nodegroups['node'])) {
-    $gids = $nodegroups['node'];
+  if (!empty($groups['node'])) {
+    $gids = $groups['node'];
   }
   else {
     $gids = array();
