diff --git a/og_menu.install b/og_menu.install
index 9356ef6..6e62f27 100644
--- a/og_menu.install
+++ b/og_menu.install
@@ -220,3 +220,11 @@ function og_menu_update_7302(&$sandbox) {
 function og_menu_update_7303(&$sandbox) {
   menu_cache_clear_all();
 }
+
+/**
+ * Clear menu cache to ensure changes in hook_menu().
+ */
+function og_menu_update_7304(&$sandbox) {
+  // No need to clear menu cache her, because all caches will be cleared anyway
+  // while finishing update process in update_finished().
+}
\ No newline at end of file
diff --git a/og_menu.module b/og_menu.module
index b099c0a..e003930 100644
--- a/og_menu.module
+++ b/og_menu.module
@@ -107,7 +107,7 @@ function og_menu_menu() {
     'page callback' => 'drupal_get_form',
     'page arguments' => array('og_menu_overview_form', 1, 2, 5),
     'access callback' => 'og_menu_access',
-    'access arguments' => array(1, 2),
+    'access arguments' => array(1, 2, 'list-menu', 5),
     'type' => MENU_CALLBACK,
     'weight' => 1,
     'file' => 'og_menu.pages.inc',
@@ -124,7 +124,7 @@ function og_menu_menu() {
     'page callback' => 'drupal_get_form',
     'page arguments' => array('og_menu_edit_item_form', 'add', 1, 2, 5),
     'access callback' => 'og_menu_access',
-    'access arguments' => array(1, 2),
+    'access arguments' => array(1, 2, 'add-menu-item', 5),
     'type' => MENU_LOCAL_TASK,
     'weight' => 1,
     'file' => 'og_menu.pages.inc',
@@ -135,7 +135,7 @@ function og_menu_menu() {
     'page callback' => 'drupal_get_form',
     'page arguments' => array('og_menu_edit_menu_form', 'edit', 1, 2, 5),
     'access callback' => 'og_menu_access',
-    'access arguments' => array(1, 2),
+    'access arguments' => array(1, 2, 'edit-menu', 5),
     'type' => MENU_LOCAL_TASK,
     'weight' => 1,
     'file' => 'og_menu.pages.inc',
@@ -146,7 +146,7 @@ function og_menu_menu() {
     'page callback' => 'drupal_get_form',
     'page arguments' => array('og_menu_delete_menu_confirm', 5, 1, 2),
     'access callback' => 'og_menu_access',
-    'access arguments' => array(1, 2),
+    'access arguments' => array(1, 2, 'delete-menu', 5),
     'type' => MENU_CALLBACK,
     'weight' => 1,
     'file' => 'og_menu.pages.inc',
@@ -157,7 +157,7 @@ function og_menu_menu() {
     'page callback' => 'drupal_get_form',
     'page arguments' => array('og_menu_edit_item_form', 'edit', 1, 2, 5, 7),
     'access callback' => 'og_menu_access',
-    'access arguments' => array(1, 2),
+    'access arguments' => array(1, 2, 'edit-menu-item', 5, 7),
     'type' => MENU_LOCAL_TASK,
     'weight' => 1,
     'file' => 'og_menu.pages.inc',
@@ -168,7 +168,7 @@ function og_menu_menu() {
     'page callback' => 'drupal_get_form',
     'page arguments' => array('og_menu_delete_item_form', 7, 1, 2, 5),
     'access callback' => 'og_menu_access',
-    'access arguments' => array(1, 2),
+    'access arguments' => array(1, 2, 'delete-menu-item', 5, 7),
     'type' => MENU_CALLBACK,
     'weight' => 1,
     'file' => 'og_menu.pages.inc',
@@ -987,11 +987,31 @@ function og_menu_get_menus($group_type, $gid) {
 /**
  * Access function.
  */
-function og_menu_access($group_type, $gid, $op = NULL) {
+function og_menu_access($group_type, $gid, $op = NULL, $menu = NULL, $menu_item = NULL) {
+  // Make sure that gid is id of an organic group.
   if (!og_is_group($group_type, $gid)) {
     return FALSE;
   }
-  elseif (user_access('administer menu')) {
+
+  // Make sure that menu, is an og_menu-menu and belongs to the given group
+  // of given group type.
+  if ($menu) {
+    $query = db_select('og_menu', 'ogm');
+    $query->condition('ogm.gid', $gid, '=');
+    $query->condition('ogm.menu_name', $menu['menu_name'], '=');
+    $query->condition('ogm.group_type', $group_type, '=');
+    $count = $query->countQuery()->execute()->fetchField();
+    if (!$count) {
+      return FALSE;
+    }
+
+    // Make sure, that menu-item is an item of the og_menu-menu.
+    if ($menu_item && $menu['menu_name'] != $menu_item['menu_name']) {
+      return FALSE;
+    }
+  }
+
+  if (user_access('administer menu')) {
     return TRUE;
   }
   elseif (user_access('administer og menu') || og_user_access_entity('administer og menu', $group_type, $gid)) {
